gokart/utils.py

```diff
@@ -71,6 +71,19 @@ def flatten(targets: FlattenableItems[T]) -> list[T]:
     return flat
 
 
+K = TypeVar('K')
+
+
+def map_flattenable_items(items: FlattenableItems[T], func: Callable[[T], K]) -> FlattenableItems[K]:
+    if isinstance(items, dict):
+        return {k: map_flattenable_items(v, func) for k, v in items.items()}
+    if isinstance(items, str):
+        return items  # type: ignore
```

Comment:
In this case, `T` means `str, so you should apply func for this.

```suggestion
    if isinstance(items, str):
        return func(items) # type: ignore
```